test/release-build-test/check-release-build.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# Copyright 2021 VMware Tanzu Community Edition contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+version="${1:?TCE version argument empty. Example usage: ./test/release-build-test/check-release-build.sh v0.10.0}"
+: "${GITHUB_TOKEN:?GITHUB_TOKEN is not set}"
+
+TCE_REPO_URL="https://github.com/vmware-tanzu/community-edition"
+OS=$(uname -s | tr '[:upper:]' '[:lower:]')
+ARCH="amd64"
+
+temp_dir=$(mktemp -d)
+
+TCE_TAR_BALL="${temp_dir}/tce-${OS}-${ARCH}-${version}.tar.gz"
+TCE_INSTALLATION_DIR="${temp_dir}/tce-${OS}-${ARCH}-${version}"
+
+gh release download "${version}" --repo ${TCE_REPO_URL} --pattern "tce-${OS}-${ARCH}-${version}.tar.gz" --dir "${temp_dir}"
+
+tar xvzf "${TCE_TAR_BALL}" --directory "${temp_dir}"
+
+if [ "${OS}" == 'darwin' ]; then
+    for binary in "${TCE_INSTALLATION_DIR}"/bin/*; do
+    spctl -vv --type install --asses "${binary}"
+    done
+fi
+
+"${TCE_INSTALLATION_DIR}"/install.sh
+
+tanzu version
+
+tanzu cluster version
+
+tanzu conformance version
+
+tanzu diagnostics version
+
+tanzu kubernetes-release version
+
+tanzu management-cluster version
+
+tanzu package version
+
+tanzu standalone-cluster version
+
+tanzu pinniped-auth version
+
+tanzu builder version
+
+tanzu login version
```

Comment:
can also check the version match for TCE plugins

Reply:
> can also check the version match for TCE plugins

@navidshaikh Makes sense from face value, there is still some complexity (more code etc) though. For example, for `standalone-cluster` plugin, the output is like this

```bash
$ tanzu standalone-cluster version
v0.9.1
```

for conformance, the output is like this

```bash
$ tanzu conformance version
Sonobuoy Version: v0.53.2
TCE Version: v0.9.1
TCE SHA 00d8d13a
```

We might probably end up using `grep` to do a simple check of the version `v0.9.1` in the above 3 line output, compared to an exact output check which could be a bit more verbose to check all versions including SHA, sonobuoy version. Both of it is possible. I'm just thinking of refraining from doing that though, as there is also a need for updating the tests before the release with the expected version when we include test for Sonobuoy version, but yeah other things can be inferred - `v0.9.1` we already have from `release_version` input and we can also find the corresponding `git` SHA for `TCE SHA`.

Also, are we only talking about the 3 TCE plugins here or even Tanzu Framework plugins? I think during one discussion or even in my though process while writing this script, there was an idea to do a version check for all of them, because - why not? Why stop at just checking if the CLI plugins work with `version` command

I think a good way to take a call in this case would be to think about the goal of the script. The goal of this part of the script is currently to simply check if all the `tanzu` CLI plugins (Tanzu Framework + TCE plugins) work by invoking the `version` command which invokes the CLI plugin binary to get the version. When I say `work` - I mean that the binary is present and installed and picked up by `tanzu` CLI and is _not_ corrupt after the binary signing process. We could also add the goal of checking if the version of the CLI plugins (TCE or both TCE + Tanzu Framework CLI plugins) are all correct

Reply:
Adding a version command output check at release gate makes sense in terms of avoiding issues like #1383 , though in the case of #1383 the `Sonobuoy SHA` was entirely removed from the output. Also, doing `version` command output check doesn't exactly make sense in `main` for non-release commits, unless we want to see if SHA was set right to the version info or `dev` version shows up, which is less priority than a released CLI version output which people use while reporting issues

Reply:
I think it makes sense to add it, let me know to what extent - like, just check for example `v0.9.1` TCE release version in the `version` command of TCE plugins or more in case of `conformance` CLI plugin and what about Tanzu Framework CLI plugins

Reply:
Sure, we can do version check in a follow up PR.